scripts/preload/win32_wrap.rb

```diff
@@ -0,0 +1,369 @@
+		:G => 0x47, :H => 0x48, :I => 0x49, :J => 0x4A, :K => 0x4B, :L => 0x4C,
+		:M => 0x4D, :N => 0x4E, :O => 0x4F, :P => 0x50, :Q => 0x51, :R => 0x52,
+		:S => 0x53, :T => 0x54, :U => 0x55, :V => 0x56, :W => 0x57, :X => 0x58,
+		:Y => 0x59, :Z => 0x5A,
+
+		:LWIN => 0x5B, :RWIN => 0x5C,
+
+		:NUMPAD0 => 0x60, :NUMPAD1 => 0x61, :NUMPAD2 => 0x62, :NUMPAD3 => 0x63,
+		:NUMPAD4 => 0x64, :NUMPAD5 => 0x65, :NUMPAD6 => 0x66, :NUMPAD7 => 0x67,
+		:NUMPAD8 => 0x68, :NUMPAD9 => 0x69,
+		:MULTIPLY => 0x6A, :ADD => 0x6B, :SEPARATOR => 0x6C, :SUBSTRACT => 0x6D,
+		:DECIMAL => 0x6E, :DIVIDE => 0x6F,
+
+		:F1 => 0x70, :F2 => 0x71, :F3 => 0x72, :F4 => 0x73,
+		:F5 => 0x74, :F6 => 0x75, :F7 => 0x76, :F8 => 0x77,
+		:F9 => 0x78, :F10 => 0x79, :F11 => 0x7A, :F12 => 0x7B,
+		:F13 => 0x7C, :F14 => 0x7D, :F15 => 0x7E, :F16 => 0x7F,
+		:F17 => 0x80, :F18 => 0x81, :F19 => 0x82, :F20 => 0x83,
+		:F21 => 0x84, :F22 => 0x85, :F23 => 0x86, :F24 => 0x87,
+
+		:NUMLOCK => 0x90, :SCROLL => 0x91,
+		:LSHIFT => 0xA0, :RSHIFT => 0xA1, :LCONTROL => 0xA2, :RCONTROL => 0xA3,
+		:LMENU => 0xA4, :RMENU => 0xA5, :OEM_1 => 0xBA,
+		:OEM_PLUS => 0xBB, :OEM_COMMA => 0xBC, :OEM_MINUS => 0xBD, :OEM_PERIOD => 0xBE,
+		:OEM_2 => 0xBF, :OEM_3 => 0xC0, :OEM_4 => 0xDB, :OEM_5 => 0xDC,
+		:OEM_6 => 0xDD, :OEM_7 => 0xDE
+	}
+
+	WIN32INV = WIN32.invert
+
+	WIN2SDL = {
+		:BACK => :BACKSPACE,
+		:CAPITAL => :CAPSLOCK,
+		:PRIOR => :PAGEUP, :NEXT => :PAGEDOWN,
+		:PRINT => :PRINTSCREEN,
+
+		:LWIN => :LGUI, :RWIN => :RGUI,
+
+		:NUMPAD0 => :KP_0, :NUMPAD1 => :KP_1, :NUMPAD2 => :KP_2, :NUMPAD3 => :KP_3,
+		:NUMPAD4 => :KP_4, :NUMPAD5 => :KP_5, :NUMPAD6 => :KP_6, :NUMPAD7 => :KP_7,
+		:NUMPAD8 => :KP_8, :NUMPAD9 => :KP_9,
+		:MULTIPLY => :KP_MULTIPLY, :ADD => :KP_PLUS, :SUBSTRACT => :KP_MINUS,
+		:DECIMAL => :KP_DECIMAL, :DIVIDE => :KP_DIVIDE,
+
+		:NUMLOCK => :NUMLOCKCLEAR, :SCROLL => :SCROLLLOCK,
+		:LCONTROL => :LCTRL, :RCONTROL => :RCTRL,
+		# FIXME: Fill these out
+		:LMENU => :LALT, :RMENU => :RALT, :OEM_1 => :SEMICOLON,
+		:OEM_PLUS => :UNKNOWN, :OEM_COMMA => :UNKNOWN, :OEM_MINUS => :UNKNOWN, :OEM_PERIOD => :UNKNOWN,
+		:OEM_2 => :UNKNOWN, :OEM_3 => :UNKNOWN, :OEM_4 => :UNKNOWN, :OEM_5 => :UNKNOWN,
+		:OEM_6 => :UNKNOWN, :OEM_7 => :UNKNOWN
+	}
+
+	WIN2SDL.default = :UNKNOWN
+end
+
+$win32KeyStates = nil
+
+module Graphics
+	class << self
+		alias_method(:win32wrap_update, :update)
+		def update
+			win32wrap_update
+			$win32KeyStates = nil
+		end
+	end
+end
+
+def get_raw_keystates
+	if $win32KeyStates == nil
+		$win32KeyStates = MKXP.raw_key_states
+	end
+
+	return $win32KeyStates
+end
+
+def common_keystate(vkey)
+	vkey_name = Scancodes::WIN32INV[vkey]
+
+	if vkey_name == :LBUTTON
+		return Input.press?(Input::MOUSELEFT) ? 1 : 0
+	elsif vkey_name == :RBUTTON
+		return Input.press?(Input::MOUSERIGHT) ? 1 : 0
+	elsif vkey_name == :MBUTTON
+		return Input.press?(Input::MOUSEMIDDLE) ? 1 : 0
+	end
+
+	offset = Scancodes::SDL[vkey_name]
+	return get_raw_keystates.getbyte(offset)
```

Comment:
```suggestion
	states = get_raw_keystates
	pressed = false

	if vkey == :LBUTTON
		pressed = Input.press?(Input::MOUSELEFT)
	elsif vkey == :RBUTTON
		pressed = Input.press?(Input::MOUSERIGHT)
	elsif vkey == :MBUTTON
		pressed = Input.press?(Input::MOUSEMIDDLE)
	elsif vkey == :SHIFT
		pressed = double_state(states, :LSHIFT, :RSHIFT)
	elsif vkey == :MENU
		pressed = double_state(states, :LALT, :RALT)
	elsif vkey == :CONTROL
		pressed = double_state(states, :LCTRL, :RCTRL)
	else
		scan = nil
		if Scancodes::SDL.key?(vkey_name)
			scan = vkey_name
		else
			scan = Scancodes::WIN2SDL[vkey_name]
		end

		pressed = state_pressed(states, scan)
	end

	return pressed ? 1 : 0
```
Okay, after bashing my head for way too long trying to figure out why key presses weren't working (note to self: `true != 1`), here's a couple more changes as well as fixing an issue that applied to the original mkxp.